docs/architecture.md

```diff
@@ -18,3 +18,94 @@
   subscribe to the Redux store without React, but this would be more complicated.
 
 ## Back-end architecture
+
+![Diagram](images/architecture-back-end.drawio.svg)
+
+### Dataset files
+
+All persistent data is stored on the back-end server as JSON files, in the following folder structure
+as seen from the SERVER_DATA_ROOT location:
+
+```
+├── datasets
+│   ├── some-dataset
+│   │   ├── locations.json (array of lng-lat coordinates for each initiative)
+│   │   ├── searchable.json (array of the property values and searchable strings for each initiative)
+│   │   ├── initiatives
+│   │   |   ├── 0.json (full info of first initiative in the above aggregate JSONs)
+│   │   |   ├── 1.json
+│   │   |   ├── ...
+│   ├── other-dataset
+│   │   ├── ...
+│   ├── ...
+```
+
+Additionally, for each map variant, there's a `config.json`, which contains config for displaying the map in the UI, including the vocabs (translations of data IDs), default sidebar panel, and popup appearance.
+
+### Example file contents
+
+`locations.json`:
+
+```
+[ [1.21419, 50.45254], [0.21002, 49.33954], … ]
+```
+
+`searchable.json`:
+
+```
+[ { "coun": "coun:GB", "sz": "sz:small", "searchString": "some co-op 2 green lane london n4 9qr" }, … ]
```

Comment:
For related reasons I'm wondering if the really bulky bit, the text `searchString` part, should be kept in its own file. This file could just be a plain text file, with one line per item. Searching it could be done by streaming it from disk, which avoids loading the entire file permanently into memory (for each dataset).

For instance, this SO [thread][1], top of the google search I just made for "javascript search text file streaming", has some sample stream-searching code, and a reference to a module which performs the streaming by what appears to be a fast non-buffering algorithm.

[1]: https://stackoverflow.com/questions/20187145/how-to-search-stream-for-string-in-node-js
[2]: https://github.com/mscdex/streamsearch

Reply:
I think these are optimisations that we don't need to make yet for the number of markers and filterable fields in the Delhi demo. The server will have enough memory. We need to prioritise getting a back-end working and not spend on time on nice-to-haves. They will add more complexity to the code rather vs just reading in the JSON straight into a `searchable` object.

I will note these optimisations in the docs for reference - the code isn't set in stone and it will be easy to make them in the future if needed.